src/components/cdPipeline/CDPipeline.tsx

```diff
@@ -1095,10 +1086,56 @@ export default class CDPipeline extends Component<CDPipelineProps, CDPipelineSta
                     </label>
                 </div>
                 {this.renderNamespaceInfo(namespaceEditable)}
+                {this.renderTriggerType()}
             </>
         )
     }
 
+    renderTriggerType() {
+      return (
+          <>
+              <p className="fs-14 fw-6 cn-9 mb-8">When do you want to deploy</p>
+              <div className="flex mb-20">
+                  <div
+                      className={`flex dc__content-start pointer w-50 pt-8 pr-16 pb-8 pl-16 br-4 mr-8 bw-1${
+                          this.state.pipelineConfig.triggerType === TriggerType.Auto ? ' bcb-1 eb-2' : ' bcn-0 en-2'
+                      }`}
+                      onClick={() => this.handleTriggerTypeChange(TriggerType.Auto)}
```

Comment:
can we extract this to a separate function?

Reply:
Done